### PR TITLE
Create GitHub Action to rebuild and commit shaders

### PR DESCRIPTION
**Pull request by startaq**
This GitHub Action will checkout the current source, run `make shaders` to rebuild the shaders and commit the updated shaders back to the repository. It's not run automatically at this point, but must be manually triggered.

Example run using this workflow: https://github.com/startaq/mame/commit/d69d1a9722b925cadefc2236abedaf7c5f4b192b

### CODE REVIEW

File: .github/workflows/rebuild-shaders.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        install: git make mingw-w64-x86_64-gcc mingw-w64-x86_64-python mingw-w64-x86_64-libc++    
+    - uses: actions/checkout@master
+    - name: Build
+      env:
+        MINGW64: "/mingw64"
+      run: make shaders
+    - uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: Rebuild shaders
+        repository: bgfx/shaders
```
Comment (mmicko) [inline .github/workflows/rebuild-shaders.yml:23]: should not this be mamedev/mame ?

Comment (startaq) [reply .github/workflows/rebuild-shaders.yml:23]: No, this specifies the directory inside the repository to check for updated files. Anything in `bgfx/shaders` that was changed will be committed.

Comment (mmicko) [reply .github/workflows/rebuild-shaders.yml:23]: Thanks, "repository" was thing that gave me wrong impression :)

**Comment by cuavas** [comment]
Can this rebuild the Metal shaders, or does that require macOS?

How would this handle credentials/authentication?

**Comment by cuavas** [comment]
Oh, one other thing – it’s not safe to do a straight make like this because of the way the generated GLSL parser source is checked in.  If for whatever reason git leaves the yacc file with a newer timestamp than the generated lexer source, it will try to rebuild it.  However, it’s not compatible with current versions of bison.  To ensure it won’t try rebuild the GLSL parser, you need to touch these files:
* 3rdparty/bgfx/3rdparty/glsl-optimizer/src/glsl/glcpp/glcpp-lex.c
* 3rdparty/bgfx/3rdparty/glsl-optimizer/src/glsl/glcpp/glcpp-parse.c
* 3rdparty/bgfx/3rdparty/glsl-optimizer/src/glsl/glcpp/glcpp-parse.h
* 3rdparty/bgfx/3rdparty/glslang/glslang/MachineIndependent/glslang_tab.cpp
* 3rdparty/bgfx/3rdparty/glslang/glslang/MachineIndependent/glslang_tab.cpp.h

**Comment by startaq** [comment]
Yes, it rebuilds the metal shaders.

Authentication is handled by the automatically provided `GITHUB_TOKEN`. Author is set to the user that manually executed the workflow and committer is set to "GitHub Actions". Example from my forked repo:

```
commit d69d1a9722b925cadefc2236abedaf7c5f4b192b (HEAD -> master, origin/master, origin/HEAD)
Author: startaq <startaq@users.noreply.github.com>
Commit: GitHub Actions <actions@github.com>

    Rebuild shaders
```

**Comment by cuavas** [comment]
I’d still rather not trust auto-commit workflow actions.  I‘ve added this in a form that uploads the rebuilt shaders as an artefact instead.

**Pull request closed without merging**